ast/compile.go

```diff
@@ -3134,13 +3134,10 @@ func (vs unsafeVars) Slice() (result []unsafePair) {
 // contains a mapping of expressions to unsafe variables in those expressions.
 func reorderBodyForSafety(builtins map[string]*Builtin, arity func(Ref) int, globals VarSet, body Body) (Body, unsafeVars) {
 
-	body, unsafe := reorderBodyForClosures(arity, globals, body)
-	if len(unsafe) != 0 {
-		return nil, unsafe
-	}
-
-	reordered := Body{}
+	bodyVars := body.Vars(SafetyCheckVisitorParams)
+	reordered := make(Body, 0, len(body))
```

Comment:
micro-optimizations of little consequence:
1. the bodyVars never change, so we can collect them outside of the loop (compared to reorderBodyForClosures)
2. we know how large reordered could become, and usually would, if there are no unsafe variables; so allocate upfront.